b2/account_info/sqlite_account_info.py

```diff
@@ -169,23 +203,31 @@ def clear(self):
             conn.execute('DELETE FROM bucket_upload_url;')
 
     def set_auth_data(
-        self, account_id, account_auth_token, api_url, download_url, minimum_part_size,
-        application_key, realm
+        self,
+        account_id,
+        account_auth_token,
+        api_url,
+        download_url,
+        minimum_part_size,
+        application_key,
+        realm,
+        allowed=None
     ):
+        assert self.allowed_is_valid(allowed)
         with self._get_connection() as conn:
             conn.execute('DELETE FROM account;')
             conn.execute('DELETE FROM bucket;')
             conn.execute('DELETE FROM bucket_upload_url;')
             insert_statement = """
                 INSERT INTO account
-                (account_id, application_key, account_auth_token, api_url, download_url, minimum_part_size, realm)
-                values (?, ?, ?, ?, ?, ?, ?);
+                (account_id, application_key, account_auth_token, api_url, download_url, minimum_part_size, realm, allowed)
+                values (?, ?, ?, ?, ?, ?, ?, ?);
```

Comment:
`update_number INT NOT NULL` and this function has a default value of `None` for allowed argument, which will cause that insert to fail. Calling a function without specifying a non-mandatory argument, so that the default is used, should not cause an unconditional exception from the database layer...

Reply:
I'm confused.  This insert statement is inserting a row with auth data. The update number is not part of this row.